include/envoy/network/filter.h

```diff
@@ -448,5 +449,12 @@ class FilterChainFactory {
                                             UdpReadFilterCallbacks& callbacks) PURE;
 };
 
+class DrainingFilterChains {
+public:
+  virtual ~DrainingFilterChains() = default;
+  virtual uint64_t getDrainingListenerTag() PURE;
+  virtual std::list<const FilterChain*>& getDrainingFilterChains() PURE;
```

Comment:
nit: `const std::list<const FilterChain*>&`

Reply:
Aha, probably this one could unlock the const method.